experimental/overhead_matching/swag/data/vigor_dataset.py

```diff
@@ -136,3 +140,29 @@ def _collate_fn(samples: list[VigorDatasetItem]):
 
     return torch.utils.data.DataLoader(dataset, collate_fn=_collate_fn, **kwargs)
 
+def get_overhead_dataloader(dataset: VigorDataset, **kwargs):
+    def _collate_fn(samples: list[tuple[int, VigorDatasetItem]]):
+        return [x[0] for x in samples], VigorDatasetItem(
+            panorama_metadata=None,
+            satellite_metadata=[x[1].satellite_metadata for x in samples],
+            panorama=None,
+            satellite=torch.stack([x[1].satellite for x in samples]),
+        )
+    
+    class OverheadVigorDataset(torch.utils.data.Dataset):
+        def __init__(self, dataset: VigorDataset):
+            super().__init__()
+            self.dataset = dataset 
+        def __len__(self):
+            return len(self.dataset._satellite_metadata)
+        def __getitem__(self, idx):
```

Comment:
I don't like this, as it could become out of date with the getitem function in VigorDataset